nixos/modules/services/networking/shadowsocks.nix

```
@@ -30,6 +30,22 @@ let
 
   configFile = pkgs.writeText "shadowsocks.json" (builtins.toJSON opts);
 
+  pkgTitleMap = {
+    "${getName pkgs.shadowsocks-libev}" = "libev";
+    "${getName pkgs.shadowsocks-rust}" = "rust";
+  };
+
+  commandsMap = {
+    "${getName pkgs.shadowsocks-libev}" = {
+      server = "ss-server";
+    };
+    "${getName pkgs.shadowsocks-rust}" = {
+      server = "ssserver";
+    };
+  };
+
+  getServiceUnitName = pkg: "shadowsocks-${pkgTitleMap.${getName cfg.package}}";
```

Looks like all the transformations on `getName` reduce to just the identity?
```suggestion
  getServiceUnitName = getName cfg.package;
```